skrub/_gap_encoder.py

```diff
@@ -445,16 +456,16 @@ def fit(self, X, y=None):
             )
         self._input_name = sbd.name(X)
         self._random_state = check_random_state(self.random_state)
+        is_null = sbd.to_numpy(sbd.is_null(X))
```

Comment:
it would remove a parameter from a couple of private functions, but I would rather avoid storing this state that is only meaningful during the execution of `fit` -- not that we also don't store X as an attribute for example, but pass it around in function calls